### DISCUSSION TITLE
Updating Kong ACL Plugin with HTTP PATCH method doesn't working as expected?

### DISCUSSION THREAD

**Discussion by marcojck** [Kong/kong#7291] [Ideas and feature requests]
### Components versions

- Kong: 1.1.2

### Summary

Using HTTP PATCH to update a service ACL groups plugin causes all groups being replaced by the contents of the request payload, whereas it was expected the new groups only be appended to the existent group list.

### Steps to reproduce

1. Create the ACL plugin in a service  with two groups in the while list:
```
$ curl -X POST http://kong:8001/services/{service}/plugins \
    --data "name=acl"  \
    --data "config.whitelist=group1" \
    --data "config.whitelist=group2" \
    --data "config.hide_groups_header=true"
```

2. Update the white list with a third group using a HTTP PATCH request:
```
$ curl -X PATCH http://kong:8001/services/{service}/plugins/{acl-plugin-id} \
    --data "config.whitelist=group3" 
```
3. Actual result (the first two groups were replaced by the third group):

```
$ curl -X GET http://kong:8001/services/{service}/plugins/{acl-plugin-id} 

{
...
"config": {
    "hide_groups_header": true,
    "blacklist": null,
    "whitelist": [
      "group3"
    ]
  }
...
}
```
4. Expected result (third group being appended to the list):

```
$ curl -X GET http://kong:8001/services/{service}/plugins/{acl-plugin-id} 

{
...
"config": {
    "hide_groups_header": true,
    "blacklist": null,
    "whitelist": [
      "group1",
      "group2",
      "group3"
    ]
  }
...
}
```

Is that the right behavior? 

Thanks!

**Comment by bungle** [comment]
The current behavior is expected. I'll mark this as a feature request.

**Comment by bungle** [comment]
For JSON we could implement `merge_patch` but for other content types, there is no standard.

See: https://tools.ietf.org/html/rfc7386
